test/Datadog.Trace.Tests/CallTarget/TaskContinuationGeneratorTests.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.ClrProfiler.CallTarget.Handlers.Continuations;
+using Xunit;
+
+namespace Datadog.Trace.Tests.CallTarget
+{
+    public class TaskContinuationGeneratorTests
+    {
+        public static TReturn OnAsyncMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, CallTargetState state)
```

Comment:
Does this get correctly called for the `TaskContinuationGenerator<TIntegration, TTarget, TReturn>` case in the unit tests? Those integrations don't have a `TReturn` argument

Reply:
Yes, the `_continuation` match this signature: https://github.com/DataDog/dd-trace-dotnet/blob/57d278e86dd81680cd2e5107543b4efdc8eef407/src/Datadog.Trace.ClrProfiler.Managed/CallTarget/Handlers/Continuations/TaskContinuationGenerator.cs#L11

For the Task case the return value will be always null: https://github.com/DataDog/dd-trace-dotnet/blob/57d278e86dd81680cd2e5107543b4efdc8eef407/src/Datadog.Trace.ClrProfiler.Managed/CallTarget/Handlers/Continuations/TaskContinuationGenerator.cs#L57

Reply:
Got it, thanks 👍🏼